h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -52,6 +52,15 @@ public static Frame sort(final Frame fr, int[] cols, int[] ascending) {
   public static Frame merge(final Frame leftFrame, final Frame riteFrame, final int leftCols[], final int riteCols[],
                             boolean allLeft, int[][] id_maps) {
     int[] ascendingL, ascendingR;
+    
+    if (leftFrame.numRows()==0 && riteFrame.numRows() > 0) { // if one frame contains zero rows, returns the other frame which may contain zero rows
+      return riteFrame;
```

Comment:
imho the output should be the empty frame, or shouldn't it?

Reply:
agree

Reply:
Please explain why you marked this as resolved, this seems incorrect

Reply:
Checked with Matt Dowle: if we are joining two frames and if one of them is empty, the return result should be empty as well.  Michalk/Honza, you guys are correct.  I made a mistake to mark this as resolved.  Sorry